kube/src/api/resource.rs

```diff
@@ -1,118 +1,83 @@
 use super::params::{DeleteParams, ListParams, Patch, PatchParams, PostParams};
-use crate::{
-    api::{DynamicResource, Meta},
-    Error, Result,
-};
+use crate::{api::Meta, Error, Result};
 
-/// A Kubernetes resource that can be accessed through the API
+/// A Kubernetes request builder
 #[derive(Clone, Debug)]
-pub struct Resource {
-    /// The API version of the resource.
-    ///
-    /// This is a composite of `Resource::GROUP` and `Resource::VERSION`
-    /// (eg "apiextensions.k8s.io/v1beta1")
-    /// or just the version for resources without a group (eg "v1").
-    /// This is the string used in the `apiVersion` field of the resource's serialized form.
-    pub api_version: String,
-
-    /// The group of the resource
-    ///
-    /// or the empty string if the resource doesn't have a group.
-    pub group: String,
-
-    /// The kind of the resource.
-    ///
-    /// This is the string used in the kind field of the resource's serialized form.
-    pub kind: String,
-
-    /// The version of the resource.
-    pub version: String,
-
-    /// The namespace if the resource resides (if namespaced)
-    pub namespace: Option<String>,
+pub struct Request<K: Meta> {
+    info: K::Info,
+    namespace: Option<String>,
```

Comment:
This is the big change. Kind of want this to not be clonable, and ephemerally created by `Api`.

Reply:
Managed to get this working!

Reply:
Any reason to disallow `Clone` here? `<&T as Clone>::clone` is basically just a pointer copy.

Reply:
Main reason was to make sure that the ephemeralness was working and there were no other users.
Given how awkward it is with lifetime parameters, I don't expect we want to make it public anymore, so can put on Clone if we need it.